swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
+
+from .item import Item
+
+
+
+# WAVE 1
 class Vendor:
-    pass
+    # constructor that is by default empty, but checks for input
+    def __init__(self, inventory = [] ):
+        if inventory :
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+    # method thats adds an item to inventory
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    # method that removes an item from inventory
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    # WAVE 2
+    # method that adds items that are the same category to a list
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+
+    # WAVE 3
+    # swaps items from one inventory to another
+    def swap_items(self, another_vendor, my_item, their_item):
+        if  my_item not in self.inventory or their_item not in another_vendor.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            another_vendor.inventory.append(my_item)
+            another_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
```

Comment:
Love the guard clause!!

However, `else` conditions are typically used for last case scenarios. Every `if` conditional doesn't need to be accompanied by an `else`. When we get into web development, else clauses are useful in scenarios where we need a 'catch all' such as handling  gibberish a user enters into a form. 

Let's say a user was trying to enter their credit card number in a form.. if the user enters `!@#AFS12334` into the form, we can have conditionals describe the input as needing to be a string type, containing numeric values, etc. and in case the input doesn't cover all the required checks we can use  `else: return "Please enter a valid credit card number"` as a catch all for invalid inputs.

Reply:
We can refactor this function like so:

```suggestion
    def swap_items(self, another_vendor, my_item, their_item):
    
        if  my_item not in self.inventory or their_item not in another_vendor.inventory:
            return False
    
        self.inventory.remove(my_item)
        another_vendor.inventory.append(my_item)
        another_vendor.inventory.remove(their_item)
        self.inventory.append(their_item)
        return True
```